plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessor.java

```diff
@@ -538,7 +562,7 @@ public Answer backupSnapshot(final CopyCommand cmd) {
                     final String[] tmp = result.split("#");
                     snapshotBackupUuid = tmp[0];
                     physicalSize = Long.parseLong(tmp[1]);
-                    finalPath = folder + File.separator + snapshotBackupUuid;
+                    finalPath = folder + File.separator + snapshotBackupUuid + ".vhd";
```

Comment:
@syed This is the way it should have been, but now that in a upgraded setup there will be mixed names with ".vhd" and without, do you think this will not cause issues ? If it does we need to come up with a plan to deal with them like a upgrade step ? I think the cleanup threads may not be able to handle the old naming convention for one.  @rhtyd